### DISCUSSION TITLE
Which API gives you the Access URLs?

### DISCUSSION THREAD

**Discussion by sehrishnaseer** [kubeapps/kubeapps#3996] [Q&A] [answered]
As all the detailed information of the installed app could be retrieved from getRelease API, but from where one can get the access URLs for the app, as I didn't find it in the getRelease API. Kindly help me through this, any help would be appreciated!
![Ss](https://user-images.githubusercontent.com/95912089/147316830-ec76bab4-ce4d-44b2-9aa9-b8830a5ad945.png)

 And from where to get Cluster IP and external IP for the service using APIs. Kindly do state which one would be helpful:
![ss1](https://user-images.githubusercontent.com/95912089/147331588-de094d83-396f-4c36-b348-d9cfe69cddaa.png)

**Comment by castelblanque** [comment, marked as answer]
hi @sehrishnaseer, thanks for participating in the Kubeapps community!

Since v2.4.2 you can find that information through the operation `GetResources` of the resources plugin.

For example, if you want to get data of the `my-wordpress` service in the default ns:

```
grpcurl -plaintext \
  -d '{"installed_package_ref": {"context": {"cluster": "default", "namespace": "default"}, "plugin": {"name": "helm.packages", "version": "v1alpha1"}, "identifier": "mywordpress"} }' \
  -H "Authorization: Bearer $TOKEN" localhost:8080 \
  kubeappsapis.plugins.resources.v1alpha1.ResourcesService.GetResources | jq -s -r '.[] | select((.resourceRef.kind=="Service") and (.resourceRef.name=="mywordpress")) | .manifest."@value"' | base64 -d | jq .spec
```

If you want more insights please take a look at [the original PR](https://github.com/kubeapps/kubeapps/pull/3756).

Hope it helps!

**Reply by sehrishnaseer** [reply]
Thank you, it was helpful.
